devito/operator/profiling.py

```diff
@@ -52,8 +50,18 @@ def __init__(self, name):
         # Python-level timers
         self.py_timers = OrderedDict()
 
+        # Instance attributes
+        self._include_dirs = []
```

Comment:
since these attributes (and the corresponding _add methods (please make them private)) are used only required by "third party profilers", you should move them into the Advisor class , or inside an intermediate class for ThirdPartyProfilers/ExternalProfilers 

because here they'll never ever be used by the standard Profiler classes

Reply:
oh, I see why now...

because you need these attributes in operator.py

Reply:
So, no changes here AFAIU